packages/remirror__extension-list/src/task-list-extension.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  ApplySchemaAttributes,
+  assertGet,
+  command,
+  CommandFunction,
+  ExtensionPriority,
+  ExtensionTag,
+  getMatchString,
+  InputRule,
+  keyBinding,
+  KeyBindingProps,
+  NamedShortcut,
+  NodeExtension,
+  NodeExtensionSpec,
+  nodeInputRule,
+  NodeSpecOverride,
+  NodeViewMethod,
+  ProsemirrorNode,
+} from '@remirror/core';
+import { ExtensionListMessages as Messages } from '@remirror/messages';
+import { NodeSelection, TextSelection } from '@remirror/pm/state';
+import { ExtensionListTheme } from '@remirror/theme';
+
+import { toggleList } from './list-commands';
+import { TaskListItemExtension } from './task-list-item-extension';
+
+/**
+ * Create the node for a task list.
+ */
+export class TaskListExtension extends NodeExtension {
+  get name() {
+    return 'taskList' as const;
+  }
+
+  createTags() {
+    return [ExtensionTag.Block, ExtensionTag.ListContainerNode];
+  }
+
+  createNodeSpec(extra: ApplySchemaAttributes, override: NodeSpecOverride): NodeExtensionSpec {
+    return {
+      content: 'taskListItem+',
+      ...override,
+      attrs: extra.defaults(),
+      parseDOM: [{ tag: 'ul', getAttrs: extra.parse }, ...(override.parseDOM ?? [])],
+      toDOM: (node) => ['ul', extra.dom(node), 0],
+    };
+  }
+
+  createNodeViews(): NodeViewMethod | Record<string, never> {
+    return (_, view, getPos) => {
+      const dom = document.createElement('ul');
+      dom.style.position = 'relative';
+
+      const pos = (getPos as () => number)();
+      const $pos = view.state.doc.resolve(pos + 1);
+
+      const parentListItemNode: ProsemirrorNode | undefined = $pos.node($pos.depth - 1);
+
+      const isFirstLevel = parentListItemNode?.type?.name !== 'listItem';
+
+      if (!isFirstLevel) {
+        const parentListItemPos: number = $pos.start($pos.depth - 1);
+
+        const spine = document.createElement('div');
+        spine.contentEditable = 'false';
+        spine.classList.add(ExtensionListTheme.LIST_SPINE);
+
+        spine.addEventListener('click', (event) => {
+          view.dispatch?.(
+            view.state.tr.setSelection(NodeSelection.create(view.state.doc, parentListItemPos - 1)),
+          );
```

Comment:
Maybe I should create a separate variable `selection`, which can also improve readability:

```ts
const selection = NodeSelection.create(view.state.doc, parentListItemPos - 1);
view.dispatch?.(view.state.tr.setSelection(selection));
```

I didn't deconstruct `view` because of two reasons:

1. Pattern like `view.state.tr` is very common in a prosemirror project.
2. `view.state.tr` (or `state.tr`) has extra information then just a simple variable `tr`: it's an empty Transaction.